ui/components/resolvers/ResolversList.js

```diff
@@ -0,0 +1,159 @@
+import React from "react";
+import { Query } from "react-apollo";
+import { Spinner, ListView } from "patternfly-react";
+import GetSchema from "../../graphql/GetSchema.graphql";
+import GetResolvers from "../../graphql/GetResolvers.graphql";
+import { wellKnownTypes } from "../../helper/GraphQLWellKnownTypes";
+import { DefaultEmptyView } from "../common/DefaultEmptyView";
+import { ResolversListItem } from "./ResolversListItem";
+import style from "./resolversList.css";
+
+const groupTypes = (types, query, mutation, subscription) => types.reduce((acc, type) => {
+    switch (type.name) {
+        case query.name:
+            acc.queries.push(type);
+            break;
+        case mutation.name:
+            acc.mutations.push(type);
+            break;
+        case subscription.name:
+            acc.subscriptions.push(type);
+            break;
+        default:
+            acc.custom.push(type);
+            break;
+    }
+    return acc;
+}, {
+    queries: [],
+    mutations: [],
+    subscriptions: [],
+    custom: []
+});
+
+const renderGeneric = (schemaId, items, text, kind, onClick) => {
+    const { name, fields } = items[0];
+    return (
+        <Query key={name} query={GetResolvers} variables={{ schemaId, type: name }}>
+            {({ loading, error, data }) => {
+                if (loading) {
+                    return <Spinner className="spinner" loading />;
+                }
+                if (error) {
+                    return error.message;
+                }
+
+                const list = fields.map(field => {
+                    field.isQuery = true;
+                    return (
+                        <ResolversListItem
+                            key={name + field.name}
+                            type={name}
+                            kind={kind}
+                            item={field}
+                            resolvers={data}
+                            onClick={onClick}
+                        />
+                    );
+                });
+
+                return (
+                    <div className={style["structure-content"]}>
```

Comment:
Just wondering about the change between this and `styles.something`, which should we use?

Reply:
Use the dot notation when possible. In this case it's not possible because JS properties can't have a `-` in their name.